File: SpedTaskTracker.Api/SpedTaskTracker.Api/Services/TaskService.cs
using SpedTaskTracker.Api.Models;
using SpedTaskTracker.Api.Repositories;
using TaskStatus = SpedTaskTracker.Api.Models.TaskStatus;

namespace SpedTaskTracker.Api.Services
{
    public class TaskService : ITaskService
    {
        private readonly ITaskRepository _taskRepository;
        public TaskService(ITaskRepository taskRepository)
        {
            _taskRepository = taskRepository;
        }
        public IEnumerable<TaskEntity> GetAll()
        {
            return _taskRepository.GetAll();
        }
        public TaskEntity? GetById(int id)
        {
            return _taskRepository.GetById(id);
        }
        public void Create(TaskEntity task)
        {
            task.DataCriacao = DateTime.UtcNow;
            task.Status = TaskStatus.Pendente;
            _taskRepository.Add(task);
        }
        public void Update(TaskEntity task)
        {

            var existingTask = _taskRepository.GetById(task.Id);

            if (existingTask == null)
                return;

            existingTask.Titulo = task.Titulo;
            existingTask.Descricao = task.Descricao;
            existingTask.Status = task.Status;

            _taskRepository.Update(existingTask);
        }
        public void Delete(int id)
        {
            var task = _taskRepository.GetById(id);

            if (task == null)
                return; 

            _taskRepository.Delete(task);
        }


    }
}

File: sped-task-tracker-frontend/src/app/core/task.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Task } from '../models/task.model';
import { TaskStatus } from '../models/task-status.model';


@Injectable({
  providedIn: 'root'
})
export class TaskService {

  private readonly apiUrl = 'http://localhost:5147/api/Task'
  // inserido porta fixa para não precisar utilizar o .env com a url



  constructor(private http: HttpClient) { }


  getAll(): Observable<Task[]> {
    return this.http.get<Task[]>(this.apiUrl)
  }


  create(task: {
    titulo: string;
    descricao: string;
  }): Observable<Task> {
    return this.http.post<Task>(this.apiUrl, task);
  }


  update(
    id: number,
    task: {
      titulo: string;
      descricao: string;
      status: TaskStatus;
    }
  ): Observable<Task> {
    return this.http.put<Task>(`${this.apiUrl}/${id}`, task);
  }






}